src/steamship/client/client.py

```diff
@@ -104,6 +106,46 @@ def embed_and_search(
             space=space,
         )
 
+    def use(
+        self, app_handle: str, handle: str = None, config: Dict[str, Any] = None
```

Comment:
I would also add the optional version parameter.

Reply:
I'm thinking if we can come up with a better naming convention for `app_handle` vs `handle`. Just reading them requires me to read the documentation, which is something we want to avoid. Can't think of a better way right now, will come back to this.

Reply:
1. `reuse` I think is a great general re-phrasing of the (misnamed) `upsert` across the codebase. I think it might make sense to do that as a separate PR in which we globally rename that variable? That will be a breaking change with respect to current plugin & app code.

2. Optional version param - agree. Will add to this PR.

3. app_handle vs handle -- I don't like that pairing either. Let's plan for a future PR, but what do you think of the following proposal:

Option 1: "Allow anything":

   * `app` is always `Union[str, App]`. If `App` we transform it to `app.id`. If `str` it can be either `handle or id`. On the engine-side, we use regex to guess if it's UUID-like and then either do the lookup against `id` or `handle` in the DB.

Option 2: "Always handle":

   * Systems like NPM, Github only ever use the handle in their API. From the customer's perspective that's all that matters. We could adopt a similar posture in which the ID field is basically an internal piece of state; the customer only ever needs to remember the human readable name.